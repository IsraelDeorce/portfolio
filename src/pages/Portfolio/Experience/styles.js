import styled from 'styled-components'

export const Experience = styled.section``

export const Projects = styled.div`
  display: flex;
  justify-content: center;

  flex-wrap: wrap;
  margin: 0 10%;
  > * {
    padding: 0 1em 1em 1em;
  }
`
