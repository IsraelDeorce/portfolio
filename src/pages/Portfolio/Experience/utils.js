import { SiHp, SiDell } from 'react-icons/si'
import { GiClapperboard, GiProgression } from 'react-icons/gi'
import pucrs from '../../../assets/pucrs.svg'
import ages from '../../../assets/ages.jpg'
export const projects = [
  {
    image: <SiHp color='#0096D6' />,
    title: 'HP Inc.',
    subtitle: 'subtitle',
    description: 'desc.hp'
  },
  {
    image: <SiDell color='#007DB8' />,
    title: 'Dell Inc.',
    subtitle: 'subtitle',
    description: 'desc.dell'
  },
  {
    image: <img src={pucrs} />,
    title: 'PUCRS',
    subtitle: 'desc.research_intern',
    description: 'desc.intern'
  },
  {
    image: <GiClapperboard color='#FF7F00' />,
    title: 'Gina Tells',
    subtitle: 'desc.personal_project',
    description: 'desc.ginatells'
  },
  {
    image: <GiProgression  color='#0E6B0E' />,
    title: 'Software KPIs',
    subtitle: 'desc.personal_project',
    description: 'desc.kpis'
  },
  {
    image: <img src={ages} />,
    title: 'AGES',
    subtitle: 'desc.nonprofit_engineer',
    description: 'desc.ages'
  }
]
