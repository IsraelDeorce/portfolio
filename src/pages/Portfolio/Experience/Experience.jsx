import { useTranslation } from 'react-i18next'
import { Project } from '../../../components'
import { projects } from './utils'
import * as S from './styles'

function Experience() {
  const { t } = useTranslation()
  return (
    <section id='experience'>
      <h2>{t('button.experience')}</h2>
      <S.Projects>
        {projects.map((p, i) =>
          <Project
            key={i}
            image={p.image}
            title={p.title}
            subtitle={t(p.subtitle)}
            description={t(p.description)}
          />
        )}
      </S.Projects>
    </section>
  )
}

export default Experience
