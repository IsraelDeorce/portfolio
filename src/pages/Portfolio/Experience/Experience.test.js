import Experience from './Experience'

describe('<Experience/>', () => {
  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<Experience />)).toMatchSnapshot()
    })
  })
})
