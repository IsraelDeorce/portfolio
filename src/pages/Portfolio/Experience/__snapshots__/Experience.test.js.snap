// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience/> Snapshot renders without crashing using shallow 1`] = `
<section
  id="experience"
>
  <h2>
    button.experience
  </h2>
  <styled.div>
    <Project
      description="desc.hp"
      image={
        <SiHp
          color="#0096D6"
        />
      }
      key="0"
      subtitle="subtitle"
      title="HP Inc."
    />
    <Project
      description="desc.dell"
      image={
        <SiDell
          color="#007DB8"
        />
      }
      key="1"
      subtitle="subtitle"
      title="Dell Inc."
    />
    <Project
      description="desc.intern"
      image={
        <img
          src="test-file-stub"
        />
      }
      key="2"
      subtitle="desc.research_intern"
      title="PUCRS"
    />
    <Project
      description="desc.ginatells"
      image={
        <GiClapperboard
          color="#FF7F00"
        />
      }
      key="3"
      subtitle="desc.personal_project"
      title="Gina Tells"
    />
    <Project
      description="desc.kpis"
      image={
        <GiProgression
          color="#0E6B0E"
        />
      }
      key="4"
      subtitle="desc.personal_project"
      title="Software KPIs"
    />
    <Project
      description="desc.ages"
      image={
        <img
          src="test-file-stub"
        />
      }
      key="5"
      subtitle="desc.nonprofit_engineer"
      title="AGES"
    />
  </styled.div>
</section>
`;
